File: 14. Longest Common Prefix/longest_common_prefix.rb
def longest_common_prefix (list)
	out = ""
	shortest_word_len = 0
	
	for i in list
		if i == list[0]
			shortest_word_len = i.length
		end
		
		if ((0 - i.length) > (-1 * shortest_word_len))
			shortest_word_len = i.length
		end
	end
	
	for letter_index in 0...shortest_word_len
		letter = ""
		common = true
		
		for word in list
			if word == list[0]
				letter = word[letter_index]
			else
				if letter != word[letter_index]
					common = false
					break
				end
			end
		end
		
		if common
			out << letter
			common = true
		end
	end
	
	return out
end

strings = ["flower", "flow", "flight"]
puts (longest_common_prefix(strings));
puts ("\n\n\n\n")

File: 14. Longest Common Prefix/longest_common_prefix.js
function longestCommonPrefix (list){
	let out = "";
	let shortestWordLen = 0;
	
	for (i of list){
		if (i == list[0]){
			shortestWordLen = i.length;
		}
		
		if ((0 - i.length) > (-1 * shortestWordLen)){
			shortestWordLen = i.length;
		}
	}
	
	for (let letter_index = 0; letter_index < shortestWordLen; letter_index++){
		let letter = "";
		let common = true;
		
		for (word of list){
			if (word == list[0]){
				letter = word[letter_index]
			} else {
				if (letter != word[letter_index]){
					common = false
					break;
				}
			}

		}
					
		if (common){
			out += letter;
			common = true;
		}
	}
	
	return (out);
}

let strings = ["flower", "flow", "flight"]
console.log (longestCommonPrefix(strings));
console.log ("\n\n\n\n")
